modules/fundamental/benches/bench-otel.rs

```diff
@@ -0,0 +1,239 @@
+                    helper.cpe = Some(rev_cpe(cpe, rev))
+                }
+                if let Some(purl) = &mut helper.purl {
+                    helper.purl = Some(rev_purl(purl.clone(), rev));
+                }
+            }
+        }
+
+        fn rev_purl(from: PackageUrl, rev: u64) -> PackageUrl {
+            let name = from.name();
+            let new_name = format!("{}-{}", name, rev);
+            PackageUrl::from_str(
+                from.to_string()
+                    .replacen(name, new_name.as_str(), 1)
+                    .as_str(),
+            )
+            .unwrap()
+        }
+        fn rev_cpe(cpe: &OwnedUri, rev: u64) -> OwnedUri {
+            let uri = format!("{:0}", cpe);
+            let mut uri = cpe::uri::Uri::parse(uri.as_str()).unwrap();
+            let x = format!("{}_{}", uri.product(), rev);
+            uri.set_product(x.as_str()).unwrap();
+            uri.to_owned()
+        }
+        fn rev_product_tree(product_tree: &mut ProductTree, rev: u64) {
+            rev_branches(&mut product_tree.branches, rev);
+            for relationships in product_tree.relationships.iter_mut() {
+                for relationship in relationships.iter_mut() {
+                    // rev_product_id(&mut relationship.full_product_name.product_id, rev);
+                    rev_product_helper(
+                        &mut relationship.full_product_name.product_identification_helper,
+                        rev,
+                    );
+                }
+            }
+        }
+        fn rev_vulnerability(vulnerability: &mut Vulnerability, rev: u64) {
+            for cve in vulnerability.cve.iter_mut() {
+                *cve = format!("{}-{}", cve, rev);
+            }
+        }
+
+        for product_tree in doc.product_tree.iter_mut() {
+            rev_product_tree(product_tree, rev);
+        }
+        for vulnerabilities in doc.vulnerabilities.iter_mut() {
+            for vulnerability in vulnerabilities.iter_mut() {
+                rev_vulnerability(vulnerability, rev);
+            }
+        }
+
+        let data = serde_json::to_vec_pretty(&doc).expect("serialize ok");
+        Ok(Bytes::from(data))
+    }
+
+    fn setup_runtime_and_ctx() -> (Runtime, Arc<TrustifyContext>) {
+        let runtime = tokio::runtime::Builder::new_multi_thread()
+            .enable_all()
+            .build()
+            .unwrap();
+        let ctx = runtime.block_on(async { TrustifyContext::setup().await });
+        (runtime, Arc::new(ctx))
+    }
+
+    async fn reset_db(ctx: &Arc<TrustifyContext>) {
+        // reset DB tables to a clean state...
+        for table in [
+            "advisory",
+            "base_purl",
+            "versioned_purl",
+            "qualified_purl",
+            "cvss3",
+            "cpe",
+            "version_range",
+            "vulnerability",
+        ] {
+            ctx.db
+                .execute_unprepared(format!("DELETE FROM {table} WHERE 1=1").as_str())
```

Comment:
I think you should be able to just drop the `WHERE 1=1`.

Reply:
:+1: